### PR TITLE
Ade add plug comparison

### PR DESCRIPTION
**Pull request by AdemolaBTheG**

### CODE REVIEW

File: src/app/services/friends.service.ts [hunk 1]
```diff
@@ -60,6 +72,11 @@ export class FriendsService {
     )
   }
 
+  getRandomProfiles(){
+
+    //returns a random amount of profiles that can be used to suggest to user
+  }
```
Comment (eligst) [inline src/app/services/friends.service.ts]: Why is this here if it's not implemented?

File: src/app/components/social-system/add/add.component.ts [hunk 2]
```diff
@@ -35,16 +38,33 @@ export class AddComponent implements OnInit {
 
       next: data =>
       { 
-          console.log("Request wurde erfolgreich gesendet")
+          console.log("Request wurde erfolgreich gesendet");
```
Comment (eligst) [inline src/app/components/social-system/add/add.component.ts:47]: Remove debug output (goes for all `console.log()`s that are not necessary in prod.

File: src/app/components/social-system/add/add.component.ts [hunk 3]
```diff
@@ -35,16 +38,33 @@ export class AddComponent implements OnInit {
...
+          this.sentRequests.push(userId);
       },
       error: (error) =>
 
       console.log(error.message)
     });
 
+   var btn = document.getElementById("requestButton"+userId);
+   btn!.innerHTML = "Pending";
+   btn?.setAttribute("disabled","disabled");
```
Comment (eligst) [inline src/app/components/social-system/add/add.component.ts:57]: You could do all of this with bindings - the Angular way, but this is fine too if you want it

File: src/app/components/social-system/add/add.component.ts [hunk 4]
```diff
@@ -57,11 +77,35 @@ export class AddComponent implements OnInit {
...
+        for (let indexI = 0; indexI < this.filteredProfiles.length; indexI++) {
+          
+          for (let indexJ = 0; indexJ < this.currentFriends.length; indexJ++) {
+            
+            if(this.filteredProfiles[indexI].user_id == this.currentFriends[indexJ].userId && this.currentFriends[indexJ].accepted){
+
+              this.filteredProfiles.splice(indexI,1);
+            }
+          }
+        }
```
Comment (eligst) [inline src/app/components/social-system/add/add.component.ts]: There's probably a better way to do this (see https://stackoverflow.com/questions/38498258/typescript-difference-between-two-arrays)

**Review by eligst — CHANGES_REQUESTED** [review]

File: src/app/components/social-system/add/add.component.ts (hunk 4 above)
Comment (AdemolaBTheG) [reply src/app/components/social-system/add/add.component.ts]: The problem in this particular case is that I'm comparing two different data types and methods like filter only the same type. Casting would be an extra effort and I'm not quite sure whether it'd be more efficient

File: src/app/components/social-system/add/add.component.ts (hunk 3 above)
Comment (AdemolaBTheG) [reply src/app/components/social-system/add/add.component.ts:57]: I'm quite unsure on how to do this since the problem is that every request has it's own button  which is being identified by the id of the request and I wouldn't know how to access it, since I'v already tried @ViewChild which was not possible

File: src/app/components/social-system/add/add.component.ts (hunk 4 above)
Comment (eligst) [reply src/app/components/social-system/add/add.component.ts]: wouldn't something like this work?  ` this.filteredProfiles = data.filter(x => !this.currentFriends.some(y => y.userId == x.user_id && y.accepted) && x.user_id != this.currenUser);`
